packages/@ourworldindata/grapher/src/scatterCharts/ScatterPlotChart.tsx

```diff
@@ -822,6 +822,14 @@ export class ScatterPlotChart
         )
     }
 
+    private validValuesForAxisDomain(property: "x" | "y"): number[] {
+        const scaleType = property === "x" ? this.xScaleType : this.yScaleType
+        const values = this.pointsForAxisDomains.map((point) => point[property])
+        return scaleType === ScaleType.log
+            ? values.filter((v) => v > 0)
+            : values
+    }
+
```

Comment:
This is a bit dangerous and should be done as a @computed property instead. The reason for this is the following: here you have a member function that accesses several mobx computed properties. But because it itself is just a function it is not going to be automatically recomputed when one of the inputs change. On top of this, the call site in `vertical/horizontalAxisPart` is in one branch of an if.

Mobx computed properties track their dependencies and cache results, but they only work correctly if the places that create derived values are wrapped in `@computed` properties and if they are read unconditionally. 

Therefore it probably makes sense to turn this function into two computed properties, one for x and one for y values. At the site of use, this property should then be read at the top of the property body so that the dependency will be tracked regardless of which of the two if blocks is executed.

Reply:
Makes perfekt sense, will do tomorrow. Thanks for explaining!